django_tenants/middleware/main.py

```diff
@@ -19,23 +20,45 @@ class TenantMainMiddleware(MiddlewareMixin):
     """
 
     @staticmethod
-    def hostname_from_request(request):
-        """ Extracts hostname from request. Used for custom requests filtering.
-            By default removes the request's port and common prefixes.
+    @lru_cache(maxsize=128)
+    def hostname_from_request_host(request_host):
         """
-        return remove_www(request.get_host().split(':')[0])
+        Extracts hostname from request host. Used for custom requests filtering.
+        By default removes the request's port and common prefixes.
+
+        Args:
+            request_host (str): The host from the request.
+
+        Returns:
+            str: The extracted hostname.
+        """
+        return remove_www(request_host.split(':')[0])
 
     def get_tenant(self, domain_model, hostname):
+        """
+        Retrieves the tenant associated with the given hostname.
+
+        :param domain_model: The domain model to query.
+        :param hostname: The hostname to look up.
+        :return: The tenant associated with the hostname.
+        """
         domain = domain_model.objects.select_related('tenant').get(domain=hostname)
```

Comment:
Is this docstring more descriptive then `get_tenant`?  It has nothing to do with your actual change so I would have left it unchanged.